### ISSUE TITLE
How to train CRFSlotFiller in Rust?

### ISSUE THREAD

**Issue by gembin** [snipsco/snips-nlu-rs#101]
How to train CRFSlotFiller in Rust?  thanks!

**Comment by adrienball** [comment]
Hi @gembin ,
The training is done with the python library [snips-nlu](https://github.com/snipsco/snips-nlu), see the documentation [here](https://snips-nlu.readthedocs.io/en/latest/).
Once you have trained an NLU engine, you can load it and run the inference with the Rust crate.

**Comment by gembin** [comment]
@adrienball yes, I'm aware training is available in the python library. Any plan to support training in rust?

**Comment by adrienball** [comment]
@gembin Implementing the training in Rust is something we have been discussing internally for the past months. It would have many advantages in terms of packaging and release process. However it requires quite some time to port the python libraries used for training only. This is not in our current short-term  roadmap, but we keep an open mind about it, and we could move forward if the trade-off with the current solution becomes favorable for a rust implementation.
I realize it is vague but I'm being transparent.
Cheers

**Comment by gembin** [comment]
Yes, totally agree, it requires some time to port.  Thank you for the detailed answer.
